modules/percolator/src/main/java/org/elasticsearch/percolator/QueryAnalyzer.java

```diff
@@ -391,42 +393,73 @@ private static Result handleDisjunction(List<Query> disjunctions, int minimumSho
         return new Result(verified, terms);
     }
 
-    static Set<QueryExtraction> selectBestExtraction(Set<QueryExtraction> extractions1, Set<QueryExtraction> extractions2) {
+    static Set<QueryExtraction> selectBestExtraction(Map<String, Float> boostFields, Set<QueryExtraction> extractions1,
+                                                     Set<QueryExtraction> extractions2) {
         assert extractions1 != null || extractions2 != null;
         if (extractions1 == null) {
             return extractions2;
         } else if (extractions2 == null) {
             return extractions1;
         } else {
+            Set<QueryExtraction> filtered1;
+            Set<QueryExtraction> filtered2;
+            if (boostFields.isEmpty() == false) {
+                Predicate<QueryExtraction> predicate = extraction -> {
+                    String fieldName = extraction.term != null ? extraction.term.field() : extraction.range.fieldName;
+                    float boost = boostFields.getOrDefault(fieldName, 1F);
+                    return boost != 0F;
+                };
+                filtered1 = extractions1.stream().filter(predicate).collect(toSet());
+                if (filtered1.isEmpty()) {
```

Comment:
The stream can only be consumed once, so if anyMatch(...) or allMatch(...) is invoked then collect(...) can no longer be invoked. So I'll keep it the way it is now...